include/psen_scan/start_request.h

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2020 Pilz GmbH & Co. KG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#ifndef PSEN_SCAN_START_REQUEST_H
+#define PSEN_SCAN_START_REQUEST_H
+
+#include <arpa/inet.h>
+#include <cstdint>
+#include <endian.h>
+#include <string>
+#include <vector>
+
+#include <psen_scan/scanner_configuration.h>
+#include <psen_scan/crc.h>
+
+#pragma pack(push, 1)
+
+namespace psen_scan
+{
+
+/**
+ * @brief Frame containing all necessary fields for a Start Request.
+ *
+ * Unless otherwise indicated the byte order is little endian.
+ *
+ */
+typedef struct StartRequest
```

Comment:
I suggest to use a class. Why?
- Our default intention is to hide things from the higher abstractions level which contradicts to some extend the default intention of a struct.
- See also: https://www.fluentcpp.com/2017/06/13/the-real-difference-between-struct-class/